File: ADS-2-Semestre/EngenhariaDeSoftware/InterfacePatoCorredor/Pato_Maratonista.java
package engsoft;


public class Pato_Maratonista extends Pato implements  Padrao_Grasnar{
    
    public Pato_Maratonista()	
	{
		
            setComportamentoCorrida(new Correr());		
	}
	
	public String mostrar() {		
		return "Eu sou o Pato Maratonista.";
	}

	public String grasnar() {
		return "Que-Que. Grrrrrrrrr.";
	}
}

File: ADS-2-Semestre/EngenhariaDeSoftware/InterfacePatoC#/Pato/PatoMaratonista.cs
using Pato;
using System;
using System.Collections.Generic;
using System.Text;

namespace JogoPato
{
    public class PatoMaratonista : Pato, IPadraoGrasnar
    {
		public PatoMaratonista()
		{

			setComportamentoCorrida(new Correr());
		}

		public String grasnar()
		{
			return "Que-Que. Grrrrrrrrr.";
		}

		public override string mostrar()
		{
			return "Eu sou o Pato Maratonista.";
		}
	}
}
